src/capstone.rs

```diff
@@ -175,29 +189,31 @@ impl Instr {
         }
 
         let detail = if decode_detail {
-            let detail = unsafe {*instr.detail};
+            let detail = unsafe { *instr.detail };
             let arch_union = detail.__bindgen_anon_1;
 
-            let arch = unsafe { match arch {
-                cs_arch::CS_ARCH_ARM        => { DetailsArch::ARM       (arch_union.arm)        },
-                cs_arch::CS_ARCH_ARM64      => { DetailsArch::ARM64     (arch_union.arm64)      },
-                cs_arch::CS_ARCH_MIPS       => { DetailsArch::MIPS      (arch_union.mips)       },
-                cs_arch::CS_ARCH_X86        => { DetailsArch::X86       (arch_union.x86)        },
-                cs_arch::CS_ARCH_PPC        => { DetailsArch::PPC       (arch_union.ppc)        },
-                cs_arch::CS_ARCH_SPARC      => { DetailsArch::SPARC     (arch_union.sparc)      },
-                cs_arch::CS_ARCH_SYSZ       => { DetailsArch::SYSZ      (arch_union.sysz)       },
-                cs_arch::CS_ARCH_XCORE      => { DetailsArch::XCORE     (arch_union.xcore)      },
-
-                #[cfg(feature = "capstone4")]
-                cs_arch::CS_ARCH_M68K       => { DetailsArch::M68K      (arch_union.m68k)       },
-                #[cfg(feature = "capstone4")]
-                cs_arch::CS_ARCH_TMS320C64X => { DetailsArch::TMS320C64X(arch_union.tms320c64x) },
-                #[cfg(feature = "capstone4")]
-                cs_arch::CS_ARCH_M680X      => { DetailsArch::M680X     (arch_union.m680x)      },
-                #[cfg(feature = "capstone4")]
-                cs_arch::CS_ARCH_EVM        => { DetailsArch::EVM       (arch_union.evm)        },
-                _ => panic!("Unexpected arch: {:?}", arch),
-            }};
+            let arch = unsafe {
+                match arch {
+                    cs_arch::CS_ARCH_ARM => DetailsArch::ARM(Box::new(arch_union.arm)),
```

Comment:
Why are we boxing this now? Is there an issue you ran into when this wasn't boxed? If not can we make this the same as every other instance?

Reply:
So basically: ARM(cs_arm) is the largest enum variant that contains 1768 bytes atleast. The second largest is: X86(cs_x86) which is only 464 bytes. Since enum size is bounded by its largest variant, having one large variant can penalize the memory layout of that enum. So Boxing that largest variant can possible reduce the enum since now to 464 bytes, but will automatically increase based on the Box variant when necessary.

Makes sense?